app/controllers/users/registrations_controller.rb

```diff
@@ -1,46 +1,43 @@
 # frozen_string_literal: true
 
 class Users::RegistrationsController < Devise::RegistrationsController
-  before_action :configure_sign_up_params, only: [:create]
-  before_action :configure_account_update_params, only: [:update]
-
-  # GET /resource/sign_up
-  # def new
-  #   super
-  # end
-
-  # POST /resource
+  protect_from_forgery with: :null_session
+  respond_to :json
   def create
-    super
+    build_resource(sign_up_params)
+    resource.save
+    sign_in(resource_name, resource)
+    render json: resource
   end
 
-  # GET /resource/edit
-  # def edit
-  #   super
-  # end
-
-  # PUT /resource
   def update
-    super
+    self.resource = resource_class.to_adapter.get!(send(:"current_#{resource_name}").to_key)
+    prev_unconfirmed_email = resource.unconfirmed_email if resource.respond_to?(:unconfirmed_email)
+
+    resource_updated = update_resource(resource, user_params)
+    yield resource if block_given?
+    if resource_updated
+      set_flash_message_for_update(resource, prev_unconfirmed_email)
+      bypass_sign_in resource, scope: resource_name if sign_in_after_change_password?
+
+      render json: resource, location: after_update_path_for(resource)
+    else
+      clean_up_passwords resource
+      set_minimum_password_length
+      respond_with resource
+    end
```

Comment:
most of this is boilerplate code from Devise! I don't fully understand it, but [from what I can gather,](https://github.com/heartcombo/devise/blob/main/app/controllers/devise/registrations_controller.rb) I think it basically creates a copy of the resource, tries to save it, and if the save is successful, gives the appropriate response and updates the resource in place, and if not, runs a few other methods to make sure the resource update fits devise's requirements.

The only reason I made this controller visible was so that I could render a json response if the resource was updated, since I was trying to use React as my frontend instead of Rails views; so the only part that is "mine" is the render json: in line 23 haha! I'd be curious your thoughts on how this controller is constructed and if there are easier ways for me to use it, if you have any.